arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ArrowVectorAccessor.java

```diff
@@ -17,79 +17,82 @@
  * under the License.
  */
 
-package org.apache.iceberg.spark.data.vectorized;
+package org.apache.iceberg.arrow.vectorized;
 
 import org.apache.arrow.vector.ValueVector;
-import org.apache.spark.sql.types.Decimal;
-import org.apache.spark.sql.vectorized.ArrowColumnVector;
-import org.apache.spark.sql.vectorized.ColumnarArray;
-import org.apache.spark.unsafe.types.UTF8String;
-
-@SuppressWarnings("checkstyle:VisibilityModifier")
-public abstract class ArrowVectorAccessor {
 
+public class ArrowVectorAccessor<DecimalT, Utf8StringT, ArrayT, ChildVectorT extends AutoCloseable> {
   private final ValueVector vector;
-  private final ArrowColumnVector[] childColumns;
+  private final ChildVectorT[] childColumns;
 
-  ArrowVectorAccessor(ValueVector vector) {
-    this.vector = vector;
-    this.childColumns = new ArrowColumnVector[0];
+  protected ArrowVectorAccessor(ValueVector vector) {
+    this(vector, null);
   }
 
-  ArrowVectorAccessor(ValueVector vector, ArrowColumnVector[] children) {
+  protected ArrowVectorAccessor(ValueVector vector, ChildVectorT[] children) {
     this.vector = vector;
     this.childColumns = children;
   }
 
-  final void close() {
-    for (ArrowColumnVector column : childColumns) {
-      // Closing an ArrowColumnVector is expected to not throw any exception
-      column.close();
+  public final void close() {
```

Comment:
how come final? Does that still make sense now that we are using thsi as a base class?

Reply:
Fixed